lib/vm.js

```diff
@@ -84,11 +84,12 @@ exports.isContext = binding.isContext;
 function sigintHandlersWrap(fn) {
   // Using the internal list here to make sure `.once()` wrappers are used,
   // not the original ones.
-  let sigintListeners = process._events.SIGINT;
-  if (!Array.isArray(sigintListeners))
-    sigintListeners = sigintListeners ? [sigintListeners] : [];
-  else
-    sigintListeners = sigintListeners.slice();
+  const sigintListeners = process._events.SIGINT;
+  const asArray = (listeners) => listeners ? [listeners] : [];
```

Comment:
I would eliminate this and make the line above an `|| []`, then, down in the conditional you can do `[listeners]`

Perhaps that is a little less clear? Not sure.